### DISCUSSION TITLE
Display: LVGL: Monochromatic: Option to vertically stack 8-bit bytes in screen buffers

### DISCUSSION THREAD

**Discussion by alen-daniel** [zephyrproject-rtos/zephyr#67568] [RFC]
## Introduction

For LVGL monochromatic displays, I am seeking thoughts on how to go about implementing vertically stacked 8-bit byte screen buffers

### Problem description

Note: I am using the word "length" to indicate the longer dimension and "width" to indicate the shorter dimention of a rectangular display.

Context: I am using an SPI monochromatic display (example: https://display.kyocera.com/tn0216anvnann-gn00) that can only write 1 line at a time along the length direction  of the LCD.

Problem: I am unable to use the Zephyr display drivers with lvgl to display in the portrait orientation (x axis is width, y axis is length of the LCD)

When I use the Zephyr display drivers for monochromatic display, this works fine if I want to display LVGL buffers in the landscape orientation, because the LVGL buffers are generated with 8-bit bytes stacked horizontally along the length direction.

But I am unable to use this to display in the portrait orientation. In portrait orientation, the LVGL buffers are generated with 8-bit bytes stacked horizontally in the width direction, but the LCD requires the buffers to be stacked vertically in the length direction.

### Proposed change

For monochromatic displays using LVGL, add an option to generate screen buffers that are 8-bit bytes stacked vertically

## Request for Comment

I seek feedback from the contributing community on how to go about doing this.

**Comment by Snevzor** [comment]
Hi @alen-daniel. 

What display driver is used by the TN0216ANVNANN-GN00? Recently, I added display rotation for the ssd16xx display controller, so I might have some suggestions.
